src/Data/Lens/Barlow.purs

```diff
@@ -1,66 +1,102 @@
 module Data.Lens.Barlow where
 
 import Prelude
-import Data.Lens (Lens')
+import Data.Lens (Optic', _Just)
 import Data.Lens.Record (prop)
+import Data.Maybe (Maybe)
+import Data.Profunctor.Choice (class Choice)
+import Data.Profunctor.Strong (class Strong)
 import Prim.Row as Row
 import Prim.Symbol as Symbol
 import Type.Prelude (class IsSymbol)
 import Type.Proxy (Proxy(..))
 
--- typelevel element 
-data TElem
-
-foreign import data Field :: Symbol -> TElem
-
 -- typelevel list
 data TList
 
 foreign import data TNil :: TList
 
-foreign import data TCons :: TElem -> TList -> TList
+foreign import data TCons :: forall k. k -> TList -> TList
+
+-- typelevel element 
+data TElem
+
+foreign import data QMark :: TElem
+
+foreign import data Field :: Symbol -> TElem
 
 class ParseSymbol (string :: Symbol) (attributes :: TList) | string -> attributes
 
 class Parse1Symbol (head :: Symbol) (tail :: Symbol) (out :: TList) | head tail -> out
 
 instance parse1Nil :: Parse1Symbol a "" (TCons (Field a) TNil)
-else instance parse1Pc ::
+else instance parse1Dot ::
   ( ParseSymbol s rest
     ) =>
   Parse1Symbol "." s (TCons (Field "") rest)
+else instance parse1QMark ::
+  ( ParseSymbol s rest
+    ) =>
+  Parse1Symbol "?" s (TCons (Field "") (TCons QMark rest))
 else instance parse1Other ::
   ( ParseSymbol s (TCons (Field acc) r)
   , Symbol.Cons o acc rest
   ) =>
   Parse1Symbol o s (TCons (Field rest) r)
 
-instance parseNil ::
+instance parseNilQMark ::
+  ParseSymbol "?" (TCons (Field "") (TCons QMark TNil))
+else instance parseNil ::
   ParseSymbol "" (TCons (Field "") TNil)
 else instance parseCons ::
   ( Symbol.Cons h t string
   , Parse1Symbol h t fl
   ) =>
   ParseSymbol string fl
 
-class ConstructBarlow (attributes :: TList) input output | attributes -> input output where
-  constructBarlow :: Proxy attributes -> Lens' input output
+class ConstructBarlow (attributes :: TList) p input output | attributes -> input output where
+  constructBarlow :: Proxy attributes -> Optic' p input output
```

Comment:
Is the only reason you use Optic' over Optic that it's already complicated enough as it is or does it yield better inference or something?

Reply:
:D Exactly!

Reply:
I need to abstract it so I can use a prism